examples/lie_labs.py

```diff
@@ -0,0 +1,87 @@
+import torch
+
+import theseus.labs.lie as lie
+import theseus.labs.lie.functional as lieF
+
+batch_size = 5
+
+# ### Lie Tensor creation functions
+g1 = lie.SE3.rand(batch_size, requires_grad=True)
+print(f"Created SE3 tensor with shape {g1.shape}")
+g2 = g1.clone()
+
+# Can create from a tensor as long as it's consistent with the desired ltype
+g3_data = lieF.so3.rand(5)
+g3 = lie.from_tensor(g3_data, lie.SO3)
+
+try:
+    x = lie.from_tensor(torch.zeros(1, 3, 3), lie.SO3)
+except ValueError as e:
+    print(f"ERROR: {e}")
+
+g4 = lie.as_lietensor(g3_data, lie.SO3)
+g5 = lie.cast(g3_data, lie.SO3)  # alias for as_lietensor
+
+# ### Lie operations
+v = torch.randn(batch_size, 6)
+
+# Exponential and logarithmic map
+out1 = lie.SE3.exp(v)  # also lie.exp(v, g1.ltype)
+print(f"Exp map returns a {type(out1)}.")
+out2 = g1.log()  # also lie.log(g1)
+print(f"Log map returns a {type(out2)}.")
+
+# Inverse
+out1 = g1.inv()  # also lie.inv(g1)
+
+# Compose
+# also lie.compose(g1, g2)
+out1 = g1.compose(g2)  # type: ignore
+
+# Differentiable jacobians
+jacs, out = g1.jcompose(g2)  # type: ignore
+print("Jacobians output is a 2-tuple.")
+print("    First element is a list of jacobians, one per group argument.")
+print(f"    For compose this means length {len(jacs)}.")
+print("    The second element of the tuple is the result of the operation itself.")
+print(f"    Which for compose is a {type(out).__name__}.")
+
+# Other options:
+#   * adj(), hat(), vee(), retract(), local(),
+#   * Jacobians: jlog(), jinv(), jexp()
+
+# ### Overriden operators
+# Compose
+out2 = g1 * g2
+torch.testing.assert_close(out1, out2, check_dtype=True)
+
+# Transfrom from (from local to world coordinate frame)
+p = torch.randn(batch_size, 3)
+pt1 = g1.transform_from(p)
```

Comment:
Would it make more sense to call this API `act` instead of `transform_from` to imply the "Action" of the group element on a vector element?

Reply:
Hmm there is [left_act and left_project](https://github.com/facebookresearch/theseus/blob/lep.labs_api_example/theseus/labs/lie/lie_tensor.py#L479). We went with `transform_from` since it is commonly used like in GTSAM but we can always alias it with `act` if this is standard anywhere else (like Sophus)?

Reply:
Just a clarification, we keep `act` for the action of Lie group elements on Lie algebra such as twists and angular velocities, where `transform`  for the action of Lie group elements on positions.

Reply:
Confirming that we leave this one as is for now and add `act` later when we include Lie algebra? @fantaosha